lib/HLSL/DxilValidation.cpp

```diff
@@ -3768,12 +3747,18 @@ static void ValidateGlobalVariables(ValidationContext &ValCtx) {
 
   if (M.GetShaderModel()->IsMS()) {
     if (TGSMSize > DXIL::kMaxMSSMSize) {
-      ValCtx.EmitFormatError(ValidationRule::SmMaxMSSMSize,
-                             { std::to_string(TGSMSize),
-                               std::to_string(DXIL::kMaxMSSMSize) });
+      Module::global_iterator GI = M.GetModule()->global_end();
+      GI--;
```

Comment:
I think you're going to want to iterate backwards until you get a global in `addrspace(3)` which is TGSM.  Otherwise, you could pick up something else in some cases, such as static global or ICB (immediate constant buffer).